library/include/rocwmma/internal/transforms_impl.hpp

```diff
@@ -93,97 +96,394 @@ namespace rocwmma
         }
     };
 
+    ///
+    /// AOS -> SOA : Transform from inline VW to ortho VW
+    ///
+    template <typename DataT, uint32_t VecSize, uint32_t... Idx>
+    ROCWMMA_DEVICE constexpr static inline auto extractEven(VecT<DataT, VecSize> const& v,
+                                                            detail::SeqT<Idx...>)
+    {
+        static_assert(sizeof...(Idx) == VecSize / 2u, "Index count must be half the vector size");
+        return VecT<DataT, VecSize / 2u>{get<Idx * 2>(v)...};
```

Comment:
Do we want to use bit shifting for multiplying and dividing VecSize by 2?

Reply:
Logically, makes sense but for readability's I didn't manually do this. Will consider if it's affecting compile time significantly